mesh/mesh-db.c

```diff
@@ -253,6 +253,134 @@ static json_object *jarray_key_del(json_object *jarray, int16_t idx)
 	return jarray_new;
 }
 
+static bool parse_imported_device_key(json_object *jobj, uint8_t key_buf[16])
+{
+	char *str = (char *)json_object_get_string(jobj);
+
+	if (!key_buf)
+		return false;
+
+	if (!str2hex(str, strlen(str), key_buf, 16))
+		return false;
+
+	return true;
+}
+
+static bool parse_imported_unicast_addr(json_object *jobj, uint16_t *unicast)
+{
+	char *str = (char *)json_object_get_string(jobj);
+
+	if (sscanf(str, "%04hx", unicast) != 1)
+		return false;
+
+	return true;
+}
+
+static bool parse_imported_net_key(json_object *jobj, uint8_t key_buf[16],
+							uint16_t *net_idx)
+{
+	json_object *jtemp, *jvalue;
+	char *str;
+	uint8_t key[16];
+
+	if (json_object_get_type(jobj) != json_type_array)
+		return false;
+
+	jtemp = json_object_array_get_idx(jobj, 0);
+
+	if (!json_object_object_get_ex(jtemp, "index", &jvalue))
+		return false;
+
+	*net_idx = (uint16_t)json_object_get_int(jvalue);
+
+	if (*net_idx > 4095)
+		return false;
+
+	if (!json_object_object_get_ex(jtemp, "key", &jvalue))
+		return false;
+
+	str = (char *)json_object_get_string(jvalue);
+	if (!str2hex(str, strlen(str), key, 16))
+		return false;
+
+	memcpy(&key_buf[0], &key[0], 16);
+
+	/* Imported node shouldn't contain oldKey or keyRefresh */
+	if (json_object_object_get_ex(jtemp, "oldKey", NULL))
+		return false;
+
+	if (json_object_object_get_ex(jtemp, "keyRefresh", NULL))
+		return false;
+
+	return true;
+}
+
+static uint32_t parse_imported_sequence(json_object *jobj,
+							uint32_t default_seq_nr)
+{
+	json_object *jvalue;
+
+	if (json_object_object_get_ex(jobj, "sequenceNumber", &jvalue))
+		return (uint32_t) json_object_get_int(jvalue);
+	else
+		return default_seq_nr;
+}
+
+static bool json_iter(json_object *jobj, struct mesh_db_import_node *i_node)
+{
+	bool dev_k = false, unicast = false;
+	bool ivi = false, ivu = false, net_k = false;
```

Comment:
Please expand names. For example `ivi` is a different entity than `iv_index`, this is confusing!